skynet/modules/stt/streaming_whisper/state.py

```diff
@@ -57,65 +56,86 @@ def _extract_transcriptions(
                 interim += word.word + space
                 log.debug(f'Participant {self.participant_id}: interim is "{interim}"')
 
         if final.strip():
             cut_mark = self.get_num_bytes_for_slicing(last_pause['end'])
             if cut_mark > 0:
                 log.debug(f'Participant {self.participant_id}: cut mark set at {cut_mark} bytes')
-                results.append(self.get_response_payload(final, True))
-                self.all_final += final
-                self.trim_working_audio(cut_mark)
-                # add 1 second of silence to the start of the working audio
-                self.working_audio = (b'\x00' * 32768) + self.working_audio
-                self.reset()
+                final_start_timestamp = self.working_audio_starts_at + int(final_starts_at * 1000)
+                final_audio = None
+                final_raw_audio = self.trim_working_audio(cut_mark)
+                if return_audio:
+                    final_audio_length = utils.convert_bytes_to_seconds(final_raw_audio)
+                    final_audio = utils.get_wav_header([final_raw_audio], final_audio_length) + final_raw_audio
+                results.append(self.get_response_payload(final, final_start_timestamp, final_audio, True))
+                # advance the start timestamp of the working audio to the start of the interim,
+                self.working_audio_starts_at += int(last_pause['end'] * 1000)
             else:
-                results.append(self.get_response_payload(final + interim))
+                # return everything as interim if failed to slice and acquire cut mark
+                results.append(
+                    self.get_response_payload(
+                        final + interim, self.working_audio_starts_at + int(ts_result.words[0].start * 1000)
+                    )
+                )
                 return results
         if interim.strip() != '':
-            results.append(self.get_response_payload(interim))
+            results.append(
+                self.get_response_payload(interim, self.working_audio_starts_at + int(interim_starts_at * 1000))
+            )
         return results
 
-    def should_transcribe(self, working_audio_duration: float) -> bool:
+    def should_transcribe(self) -> bool:
         # prevents hallucinations at the start of the audio
         if self.silent_chunks == self.chunk_count:
             return False
+        working_audio_duration = utils.convert_bytes_to_seconds(self.working_audio)
         if working_audio_duration >= 1 and not self.long_silence:
             return True
         return False
 
-    async def process(self, chunk: bytes) -> List[utils.TranscriptionResponse] | None:
+    async def process(self, chunk: bytes, chunk_timestamp: int) -> List[utils.TranscriptionResponse] | None:
         self.chunk_count += 1
-        chunk_duration = utils.convert_bytes_to_seconds(chunk)
-        self.user_time += chunk_duration
+        if self.chunk_duration == 0:
```

Comment:
This reads weird because we are not taking the duration straight from the chunk itself.

Reply:
added a chunk class